src/field/legend_cutscene_camera.c

```diff
@@ -0,0 +1,987 @@
+        flapSfx = SEQ_SE_GS_RUGIA_HABATAKI;
+    } else {
+        species = SPECIES_HO_OH;
+        mapObjectId = obj_D17R0110_hou_obj01;
+        firstFlapDelay = 0;
+        flapSfx = SEQ_SE_GS_HOUOU_HABATAKI;
+    }
+
+    mapObject = GetMapObjectByID(fieldSystem->mapObjectManager, mapObjectId);
+    GF_ASSERT(mapObject != NULL);
+
+    switch (*pState) {
+    case BIRD_FINAL_APPROACH_TASK_STATE_INIT:
+        GF_ExpHeap_FndInitAllocator(&taskData->allocator, HEAP_ID_4, 0x20);
+        Field3dModel_LoadFromFilesystem(&taskData->model, NARC_demo_legend, NARC_legend_legend_00000034_NSBMD, HEAP_ID_4);
+        Field3dModelAnimation_LoadFromFilesystem(&taskData->anims[0], &taskData->model, NARC_demo_legend, NARC_legend_legend_00000035_NSBCA, HEAP_ID_4, &taskData->allocator);
+        Field3dModelAnimation_LoadFromFilesystem(&taskData->anims[1], &taskData->model, NARC_demo_legend, NARC_legend_legend_00000036_NSBMA, HEAP_ID_4, &taskData->allocator);
+        Field3dModelAnimation_LoadFromFilesystem(&taskData->anims[2], &taskData->model, NARC_demo_legend, NARC_legend_legend_00000037_NSBTP, HEAP_ID_4, &taskData->allocator);
+        Field3dObject_InitFromModel(&taskData->object, &taskData->model);
+        Field3dObject_AddAnimation(&taskData->object, &taskData->anims[0]);
+        Field3dObject_AddAnimation(&taskData->object, &taskData->anims[1]);
+        Field3dObject_AddAnimation(&taskData->object, &taskData->anims[2]);
+        Field3dObject_SetActiveFlag(&taskData->object, FALSE);
+        modelAnimListSetFrameIndex(taskData->anims,3, 0);
+    {
+        VecFx32 pos;
+        MapObject_GetPositionVec(PlayerAvatar_GetMapObject(fieldSystem->playerAvatar), &pos);
+        if (taskData->gameVersion == VERSION_HEARTGOLD) {
+            pos.y += FX32_CONST(40);
+            pos.z -= FX32_CONST(16);
+        } else {
+            pos.z -= FX32_CONST(280);
+        }
+        Field3dObj_SetPosEx(&taskData->object, pos.x, pos.y, pos.z);
+    }
+        ++(*pState);
+        break;
+    case BIRD_FINAL_APPROACH_TASK_STATE_CORRECT_POS:
+    {
+        VecFx32 pos;
+        MapObject_GetPositionVec(mapObject, &pos);
+        taskData->yEnd = pos.y;
+        if (taskData->gameVersion == VERSION_HEARTGOLD) {
+            taskData->yDelta = FX32_CONST(120);
+        } else {
+            taskData->yDelta = FX32_CONST(160);
+        }
+        pos.y += taskData->yDelta;
+        MapObject_SetPositionVec(mapObject, &pos);
+    }
+        MapObject_SetVisible(mapObject, FALSE);
+        taskData->delayTimer = 0;
+        taskData->betweenFlapsDelayTimer = 0;
+        ++(*pState);
+        break;
+    case BIRD_FINAL_APPROACH_TASK_STATE_FLAP_DOWN:
+        ++taskData->delayTimer;
+        if (taskData->delayTimer >= firstFlapDelay) {
+            if (taskData->betweenFlapsDelayTimer == 0) {
+                PlaySE(flapSfx);
+                taskData->betweenFlapsDelayTimer = 24;
+            }
+            --taskData->betweenFlapsDelayTimer;
+        }
+    {
```

Comment:
are the scope brackets necessary to match?

Reply:
they're necessary to declare variables inside a switch case, and these `pos` occupy different regions of the stack from each other